android/examples/todo-sync/src/main/java/com/mongodb/stitch/android/examples/todosync/TodoListActivity.java

```diff
@@ -306,30 +345,43 @@ private void showEditItemDialog(final ObjectId itemId, final String currentTask)
   }
 
   private void updateTodoItemTask(final ObjectId itemId, final String newTask) {
+    final BsonObjectId docId = new BsonObjectId(itemId);
     items.sync().updateOneById(
-        new BsonObjectId(itemId),
+        docId,
         new Document("$set", new Document(TodoItem.TASK_KEY, newTask)))
-        .addOnSuccessListener(result -> todoAdapter.updateItems(getItems()))
-        .addOnFailureListener(e -> Log.e(TAG, "failed to insert log item", e));
+        .addOnSuccessListener(result -> {
+          items.sync().find(new BsonDocument("_id", docId)).first()
+              .addOnSuccessListener(doc -> {
+                if (doc == null) {
+                  return;
+                }
+                if (!TodoItem.isTodoItem(doc)) {
+                  return;
+                }
+                todoAdapter.updateOrAddItem(new TodoItem(doc));
+              })
+              .addOnFailureListener(e -> Log.e(TAG, "failed to find todo item", e));
+        })
+        .addOnFailureListener(e -> Log.e(TAG, "failed to insert todo item", e));
   }
 
   private void addTodoItem(final String task) {
     final Document newItem =
         new Document()
-            .append(TodoItem.OWNER_ID, Stitch.getDefaultAppClient().getAuth().getUser().getId())
+            .append(TodoItem.OWNER_ID, userId)
             .append(TodoItem.TASK_KEY, task)
             .append(TodoItem.CHECKED_KEY, false);
     items.sync().insertOneAndSync(newItem)
         .addOnSuccessListener(result -> {
-          todoAdapter.updateItems(getItems());
+          todoAdapter.updateOrAddItem(new TodoItem(newItem));
           touchList();
         })
-        .addOnFailureListener(e -> Log.e(TAG, "failed to insert log item", e));
+        .addOnFailureListener(e -> Log.e(TAG, "failed to insert todo item", e));
   }
 
   private void touchList() {
     lists.sync().updateOneById(
-        new BsonString(Stitch.getDefaultAppClient().getAuth().getUser().getId()),
+        new BsonString(userId),
```

Comment:
[q] Is it common to be using the raw Bson types instead of the Java-coerced ones? Seems like we should be using `String` instead of `BsonString` wherever we can. Probably a comment for GA though.

Reply:
Well we require updateOneById to accept a BsonValue. We could make it accept an Object and hope for the best that it serializes. I'm open to changing it. @adamchel ? I just went with type safety initially.

Reply:
Hmm, I think type safety is important, let's see how people feel about this during beta.